src/modules/core/components/Numeral/Numeral.tsx

```diff
@@ -25,47 +24,52 @@ export interface Props extends HTMLAttributes<HTMLSpanElement> {
   /** Suffix the value with this string */
   suffix?: string;
 
-  /** Number of decimals to show after comma */
-  truncate?: number;
+  /** Number of mantissa digits to show */
+  mantissa?: number;
+
+  /** Total length of number to show */
+  totalLength?: number;
 
   /** Number of decimals to format the number with, or unit from which to determine this (ether, gwei, etc.) */
   unit?: number | string;
 
   /** Actual value */
-  value: number | string | BigNumber;
-
-  /** Should large number be truncate to 5 figures */
-  reducedOutput?: boolean;
+  value: string | BigNumber | number;
 }
 
 const Numeral = ({
   appearance,
   className,
   prefix,
   suffix,
-  truncate,
+  mantissa,
+  totalLength,
   unit,
   value,
-  reducedOutput = true,
   ...props
 }: Props) => {
-  const formattedNumber = numberFormatter({
+  // formattedNumber could contain HTML,
+  // use outputRef to reference as an html object
+  const outputRef = useRef<HTMLSpanElement>(null);
+  const formattedNumber = numberDisplayFormatter({
     unit,
     value,
-    prefix,
-    suffix,
-    truncate,
-    reducedOutput,
+    mantissa,
+    totalLength,
   });
 
-  return (
-    <span
-      className={className || getMainClasses(appearance, styles)}
-      {...props}
-    >
-      {formattedNumber}
-    </span>
-  );
+  useEffect(() => {
+    if (outputRef.current) {
+      outputRef.current.innerHTML = `${prefix ? `${prefix} ` : ''}
+        ${formattedNumber}
```

Comment:
I really like how you cleaned  up adding the spaces and moved it all centrally here.